### PR TITLE
CI: Use pkg_resources to list installed dependencies

### PR DESCRIPTION
**Pull request by effigies**
While reading outputs in #2224, it looked like scikit-learn was not installed, but this is because the import is `sklearn`.

[Example](https://travis-ci.org/nilearn/nilearn/jobs/615268943):

```
Dependencies versions
  numpy: 1.15.4 from /home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/numpy/__init__.py
  scipy: 1.3.2 from /home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/scipy/__init__.py
  scikit-learn: not installed
  joblib: 0.14.0 from /home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/joblib/__init__.py
  matplotlib: not installed
  nibabel: 3.0.0rc1 from /home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/nibabel/__init__.py
```

I looked around, and the `pkg_resources` package that comes with `setuptools` can retrieve information about installed packages based on their package name.

Was quick to do, so coded it up rather than start an issue first. New output locally looks like:

```
========================================================================================================================
Python 3.7.5 (default, Oct 25 2019, 10:52:18) 
[Clang 4.0.1 (tags/RELEASE_401/final)]
from: /System/Volumes/Data/anaconda3/envs/nilearn-dev/bin/python

Dependencies versions
  numpy: 1.17.4 installed in /System/Volumes/Data/anaconda3/envs/nilearn-dev/lib/python3.7/site-packages
  scipy: 1.3.2 installed in /System/Volumes/Data/anaconda3/envs/nilearn-dev/lib/python3.7/site-packages
  scikit-learn: 0.22rc2.post1 installed in /System/Volumes/Data/anaconda3/envs/nilearn-dev/lib/python3.7/site-packages
  joblib: 0.14.0 installed in /System/Volumes/Data/anaconda3/envs/nilearn-dev/lib/python3.7/site-packages
  matplotlib: not installed
  nibabel: 3.0.0rc1 installed in /System/Volumes/Data/anaconda3/envs/nilearn-dev/lib/python3.7/site-packages
========================================================================================================================
```

### CODE REVIEW

**Comment by kchawla-pi** [comment]
Super cool. Thanks Chris. The Travis CI failure is unrelated. I'm working on a fix for that.

**Comment by kchawla-pi** [comment]
Pull from `master`. Also, what does it look like if a package is missing (the except bock)?

File: continuous_integration/show-python-packages-versions.py [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
 
 def print_package_version(package_name, indent='  '):
     try:
-        package = __import__(package_name)
-        version = getattr(package, '__version__', None)
-        package_file = getattr(package, '__file__', )
-        provenance_info = '{0} from {1}'.format(version, package_file)
-    except ImportError:
+        dist = pkg_resources.get_distribution(package_name)
+        provenance_info = '{0} installed in {1}'.format(dist.version,
```
Comment (kchawla-pi) [inline continuous_integration/show-python-packages-versions.py]: This should be in the `else` block. `try` should only have the statement that may generate the exception we're looking for.

**Comment by kchawla-pi** [comment]
Hey other than docs (which are a little _too_ extensive to make this easy to spot), what resource did you use to find out that this could be a useful application for this situation?

**Comment by effigies** [comment]
Hmm. I think I had a vague recollection that I'd seen something useful in `pkg_resources` before. I initially had thought it was in `packaging`.

But I can't really recall my thought process in full.

**Pull request merged**